gstools/transform/field.py

```diff
@@ -67,6 +69,56 @@ def binary(fld, divide=None, upper=None, lower=None):
         fld.field[fld.field <= divide] = lower
 
 
+def discrete(fld, values, thresholds=None):
+    """
+    Discrete transformation.
+
+    After this transformation, the field has only `len(values)` discrete
+    values.
+
+    Parameters
+    ----------
+    fld : :any:`Field`
+        Spatial Random Field class containing a generated field.
+        Field will be transformed inplace.
+    values : :any:`np.ndarray`
+        The discrete values the field will take
+    thresholds : :class:`numpy.ndarray`, optional
+        the thresholds, where the value classes are separated
+        Default: mean of the neighbouring values
+    """
+    if fld.field is None:
+        print("discrete: no field stored in SRF class.")
+    else:
+        if thresholds is None:
+            # just in case, sort the values
+            values = np.sort(values)
```

Comment:
What if I want the values to BE unsorted? Why do the given values need to be in relation to the values of the given field?

Reply:
Well, at least for the way I implemented the transformation, the values have to be monotonically increasing. I'm also not sure of how to interpret non monotonically increasing values.